src/DataGenerator/Generator/Bulk/AbstractBulkGenerator.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+/*
+ * This file was created by developers working at BitBag
+ * Do you need more information about us and what we do? Visit our https://bitbag.io website!
+ * We are hiring developers from all over the world. Join us and start your new, exciting adventure and become part of us: https://bitbag.io/career
+*/
+
+declare(strict_types=1);
+
+namespace BitBag\SyliusVueStorefront2Plugin\DataGenerator\Generator\Bulk;
+
+use BitBag\SyliusVueStorefront2Plugin\DataGenerator\ContextModel\Bulk\BulkContextInterface;
+use BitBag\SyliusVueStorefront2Plugin\DataGenerator\ContextModel\ContextInterface;
+use BitBag\SyliusVueStorefront2Plugin\DataGenerator\Exception\InvalidContextException;
+use BitBag\SyliusVueStorefront2Plugin\DataGenerator\Generator\Entity\GeneratorInterface;
+use DateTime;
+use Doctrine\ORM\EntityManagerInterface;
+
+abstract class AbstractBulkGenerator implements BulkGeneratorInterface
+{
+    private EntityManagerInterface $entityManager;
+
+    private GeneratorInterface $generator;
+
+    public function __construct(
+        EntityManagerInterface $entityManager,
+        GeneratorInterface $generator,
+    ) {
+        $this->entityManager = $entityManager;
+        $this->generator = $generator;
+    }
+
+    public function generate(ContextInterface $context): void
```

Comment:
I would spec this method, as there is some logic.

Reply:
It's an abstract class, so I wrote specs for children's implementations of this method.